### ISSUE TITLE
Support local images for Jupyter Notebooks

### ISSUE THREAD

**Issue by hamelsmu** [fastai/fastpages#72]
Locally embedded images should work for Jupyter

Example:
```md
# Chart From Local

![title](./chart.png)
```

**Comment by drscotthawley** [comment]
@hamelsmu  I wonder, if one puts one's local images in `_notebooks/images/`, does it work?  That's where the notebook images go in nbdev: https://github.com/fastai/nbdev/tree/master/nbs/images

I'm able to build my notebook (from which I removed the embedded image object) _locally_ on my laptop using the latest nbdev.export2html and running fastpages' 

```
$ python _action_files/nb2post.py 
```
but for some reason when the job gets submitted to GitHub pages (which, my Dockerfile pulls the latest nbdev), the nbdev conversion is throwing an error:
```
Traceback (most recent call last):
  File "/fastpages/nb2post.py", line 37, in <module>
    export2html.notebook2html(fname='_notebooks/*.ipynb', dest='_posts/', template_file='_action_files/fastpages.tpl')
  File "/nbdev/nbdev/export2html.py", line 435, in notebook2html
    raise Exception(msg + '\n'.join([f.name for p,f in zip(passed,files) if not p]))
Exception: Conversion failed on the following:
2019-12-21-PCA-From-Scratch.ipynb
```
https://github.com/drscotthawley/devblog2/runs/454208567?check_suite_focus=true

**Comment by hamelsmu** [comment]
A fix is coming for this soon:  https://github.com/fastai/fastpages/tree/fix-local-files-jupyter

Just waiting on nbdev release to be cut ( I have asked )

**Comment by drscotthawley** [comment]
On that note: 
I created a new version of the blog (https://github.com/drscotthawley/devblog3) using your "clicking on this link." method of cloning which triggers the initial build action and sends the PR with further instructions, and that's GREAT!

BUT, this version is not incorporating any changes to my [Dockerfile](https://github.com/drscotthawley/devblog3/blob/master/_action_files/Dockerfile), in which I'd changed to pull the latest nbdev -- it's using some other Dockerfile with just "pip install nbdev" which is commented out in my veersion.   ...Consequently the newer blog won't build. 

The previous blog I made, (https://github.com/drscotthawley/devblog2) I'd just made by forking the fastpages repo, because.....the "click on this link" method just wasn't something I was familiar with.  In that version, changes to the Dockerfile *were* used, which is why I was able to get that blog to build. 

So, my question is: Is ignoring people's changes to their Dockerfile in the 'click here'-generated blogs *intentional*, or is this a *mistake* that can easily be fixed?    Because I'd like to keep working on this, but the (newer) blog fails every time.

**Comment by hamelsmu** [comment]
Just make commits over the automatically generated PR or make your own
changes afterwards

**Comment by drscotthawley** [comment]
Uh....But what I'm saying is, that what I **did** --  I made my own changes afterwards --- and it didn't **work**. 

So, to be specific, compare https://github.com/drscotthawley/devblog3/blob/master/_action_files/Dockerfile
which has "pip install nvdev" replaced with a call to pull the latest nvdev source,

With actual the log of run: https://github.com/drscotthawley/devblog3/runs/456510000?check_suite_focus=true  in which it's still using "pip install nbdev".

**Comment by hamelsmu** [comment]
Oh yeah changes to the Dockerfile won’t work by itself you have to also
change .github/workflows/ci.yaml to reference ./ instead of fastpages@master

**Comment by drscotthawley** [comment]
Ok thanks!   ...Uh, I can't get that modification to work.   
So far I've tried replacing fastpages@master with ./, or replacing the whole line with my repository,...

```yaml
    - name: convert notebooks and word docs to posts
      #uses: fastai/fastpages@master # use the code in this repo instead of fastai/fastpages@master, for testing.
      uses: fastai/./ 
      #uses: drscotthawley/devblog3@master
```

...both of these builds fail rather quickly.   Sorry to bug you on this, but.. How does one do this properly?

**Comment by hamelsmu** [comment]
You don’t need fastai just uses: ./

**Comment by drscotthawley** [comment]
Thanks again. If I use just ./, I get...
```
##[error]Can't find 'action.yml', 'action.yaml' or 'Dockerfile' under '/home/runner/work/devblog3/devblog3'. Did you forget to run actions/checkout before running your local action?
```

**Comment by hamelsmu** [comment]
LOL copy it from fastpages

**Comment by hamelsmu** [comment]
Actually just fork the repo and follow the manual instructions in fastpages
_manual_instructions.md or something like that

**Comment by drscotthawley** [comment]
Ok, yea, I was just typing this and was about to hit return. 
...So the "devblog3" was created (via the "click this link" method) without any action.yml file.  I downloaded the one from fastai/fastpages and manually added it to my repo.

**Comment by drscotthawley** [comment]
Green light! :-) It all builds! :+1:   Thanks!~ 
https://github.com/drscotthawley/devblog3/runs/456862721?check_suite_focus=true 
https://drscotthawley.github.io/devblog3/

...except one of my notebooks mysteriously does not appear on the list:
https://github.com/drscotthawley/devblog3/blob/master/_notebooks/2017-02-23-Following-Gravity.ipynb   
Never got built.  No error messages,  In fact nbdev shows "https://github.com/drscotthawley/devblog3/blob/master/_notebooks/2017-02-23-Following-Gravity.ipynb"
.... well, maybe that's for another for another Issue, "Built notebook doesn't appear on blog"

**Comment by hamelsmu** [comment]
can you try setting bool_markdown to true pet this example
https://github.com/fastai/fastpages#faq

And let me know what happens?  Thanks

**Comment by drscotthawley** [comment]
Sure.  Action is running now...

Also, I just realized that, given how 'sensitive' nbdev is, in that it breaks a lot of things in otherwise-working notebooks:
 
This is the only notebook I have in which the title starts with a pair of quotes.  as in
```
# "Following Gravity" ML Foundations Part I
```
I wonder if the quotes break something.   After this action completes, I could try removing the quotes and see if that fixes it.

**Comment by drscotthawley** [comment]
Ok, action is finished.  In my `gh-pages` branch, under 2017/02, there is no version of the corresponding notebook (for Feb 23).  It's not there. 
https://github.com/drscotthawley/devblog3/tree/gh-pages/2017/02
Will try removing quotes from title and re-run.

**Comment by drscotthawley** [comment]
Removing quotes had no effect.   
bool_markdown is true.  
gh-pages branch still has no version of the file in question.

**Comment by drscotthawley** [comment]
...Nope, still no effect.  The file just vanishes! 
Here's the Action log.  I don't see any errors!  Weird. 
https://github.com/drscotthawley/devblog3/runs/456920267?check_suite_focus=true

**Comment by hamelsmu** [comment]
Means it’s an error in nbdev

**Comment by hamelsmu** [comment]
@drscotthawley it is a problem with your notebook, you have malformed yaml at the top of that notebook.  You will notice this error if trying to debug locally.  your title cannot have a colon.  

P.S. if you have one off questions that are not keeping with this issue's theme, its best to post on teh forum and tag me with your questions.  Otherwhise it pollutes the issue and becomes confusing for someone else searching for answers.  No big deal, just thought I would say this now.  Thanks for providing so many test cases for me!

**Comment by drscotthawley** [comment]
Understood.  Thanks.  Yea, I happened upon the idea that that file might be either corrupted or just out of date, so I uploaded it colab and then exported a fresh .ipynb file, and the result seems to appear on the list of posts. 

...I'll take up other problems (not related to this thread) on the Forum and tag you.

**Comment by drscotthawley** [comment]
Back to the topic of this thread, an update:
Contrary to what I may have said above, local images in notebooks are still *not* loading with fastpages, whether the images are stored in `_notebooks/images/` or in the main fastpages `images/` directory. 

Either location *looks* fine when running the notebooks in Jupyter, but 
* having the images in `_notesbooks/images` generates no build errors but simply...doesn't work in the end; the built blog page ends up with an undefined image link.
* putting the notebook's images in `images/` results in a build error about Posix paths from nbdev:

```
PosixPath('/github/workspace/_notebooks/../images/PCA_Eigenvector.svg') and PosixPath('/github/workspace/docs/../images/PCA_Eigenvector.svg') are the same file
converting: _notebooks/test.ipynb
Traceback (most recent call last):
  File "/fastpages/nb2post.py", line 36, in <module>
    export2html.notebook2html(fname='_notebooks/*.ipynb', dest='_posts/', template_file='/fastpages/fastpages.tpl')
  File "/usr/local/lib/python3.7/site-packages/nbdev/export2html.py", line 435, in notebook2html
    raise Exception(msg + '\n'.join([f.name for p,f in zip(passed,files) if not p]))
Exception: Conversion failed on the following:
```

I suspect nbdev wants the images in `_notebooks/images`  since fastai tends to use `_nbs/` for notebooks and `nbs/images/` for images in the notebooks.   

If someone (@hamelsmu or a motivated Fastai helper) would add a bit of action code to copy (or symbolic-ly link?) images from `_notebooks/images/` into each `YYYY/MM/DD/images/` folder for each post, I think everything would work.  
I would do this myself but I don't see `_notebooks/` mounted on the `gh-pages` branch, so... I don't know how to access the images.

**Comment by hamelsmu** [comment]
This will be fixed in an upcoming or sorry but just have to wait
